### ISSUE TITLE
Question about pre-training language model

### ISSUE THREAD

**Issue by dy1998** [FangShancheng/ABINet#43]
Hello, I find the setting of epoch in pretrain_language_model.yaml is 80, and I use 4 titan xp to pre-train language model. However, 1 epoch takes 8 hours, and 80 epochs need 640 hours(nearly 26 days). Should I have to train 80 epochs? How to judge the training of language model become convergence?

**Comment by Jack-Lee-NULL** [comment]
> Hello, I find the setting of epoch in pretrain_language_model.yaml is 80, and I use 4 titan xp to pre-train language model. However, 1 epoch takes 8 hours, and 80 epochs need 640 hours(nearly 26 days). Should I have to train 80 epochs? How to judge the training of language model become convergence?

As I reproduced, 1~2 epochs are enough and final model (vision model + language model we pretrained + alignment model)  could catch up with the performance author published in paper. But another problem I found, param use_sm is set False in pretrain_language_model.yaml. Is spelling mutation used when pretrain language model ?

**Comment by Jack-Lee-NULL** [comment]
@baudm I did. Different training methods I tried. On different dataset(MJ+ST lexicon, Wiki103), similar conclusion I got. 
1. feed incorrect word to language model when training, whose performance exceeds paper mentioned;
2. feed correct word to language model when training, whose performance is similar as paper mentioned;
@FangShancheng I test the performance of language model weight you published and contrast with I reproduced. I guess that  training method 2(as I mentioned above) is used in paper. I don't realize, it looks like that training method 1 is also reasonable.

**Comment by baudm** [comment]
@Jack-Lee-NULL what's the metric you're using for evaluation? When using the pretrained LM weights, I'm getting unexpected results (see previous comment). Did you try to check the actual individual outputs of the LM? Here's my [minimal test script](https://gist.github.com/baudm/6d2bb86290b0ef30685b3288503879f6) for checking individual inputs.

**Comment by baudm** [comment]
@FangShancheng @Jack-Lee-NULL I just checked Table 4 of the paper and word accuracy of BCN is indeed just above 40%. So I guess the results I posted in my previous comments were expected.
